### ISSUE TITLE
[BUG] cdef tuple assignment does not work during the definition

### ISSUE THREAD

**Issue by mrakgr** [cython/cython#3986]
**Describe the bug**
```
cpdef (int,int,float) make(int a, int b, float c):
    cdef (int,int,float) x = a,b,c
    return x
```
```
Error compiling Cython file:
------------------------------------------------------------
...
cpdef (int,int,float) make(int a, int b, float c):
    cdef (int,int,float) x = a,b,c
                            ^
------------------------------------------------------------

testm.pyx:2:29: Cannot assign type 'int' to '(int, int, float)'
```

The above fails with the above error.

```
cpdef (int,int,float) make(int a, int b, float c):
    cdef (int,int,float) x
    x = a,b,c
    return x
```

This on the other hand works.

**Expected behavior**
That the assignment should succeed during cdef.

**Environment (please complete the following information):**
 - OS: Win 10
 - Python version: 3.8.5
 - Cython version: 0.29.21

**Comment by mrakgr** [comment]
Let me just note that this is not the only wonky tuple assignment thing. `cdef object x = a,b,c` will raise an error as well. When I opened this issue, I had not realized that type definitions in if statements are not allowed and have since then factored out type definitions from their uses in [Spiral](https://github.com/mrakgr/The-Spiral-Language)' s Cython backend. So this issue is not problematic for me personally.

**Comment by scoder** [comment]
Does this work? 
```cython
cdef (int,int,float) x = (a,b,c)
```

**Comment by mrakgr** [comment]
Yes. It seems the `cdef` parser is different from the regular one and requires the parentheses around tuples. I actually had not realized it until you pointed it out.

**Comment by scoder** [comment]
Yes, it needs to support C-ish syntax like
```cython
cdef int x = 1, y = 2
```
which probably just gets in the way here.

**Comment by scoder** [comment]
Not sure if this is worth fixing, given that a more readable/explicit alternative syntax exists.
OTOH, it diverges from Python, so it is a bug. I'll leave it open in case someone wants to give it a try. It'd better not have too big an import on the parser, though. `cdef` parsing isn't exactly a trivial piece of code.
